tests/integration/kafka/test_kafka_integration.py

```diff
@@ -65,12 +69,25 @@ def test_produce_consume_with_ser_de_schema_registry(
     serializer = KafkaSerializer(
         client, schema_naming_strategy=record_name_strategy
     )
+
+    # jsonschema has no fqn, so we use topic_name_strategy for it
+    # (which also requires a separate producer)
```

Comment:
Interesting. I'd assumed jsonschema had some kind of record name property like Avro does. I suppose this means the naming strategies are actually an Avro-specific feature, not something applicable to the whole serializer. Not sure how to change the API so it's still straightforward to set the naming strategy for Avro, without being misled into thinking the same strategies can apply to json too. Maybe just prefix the property with `avro_` will do...

Could you describe what your use case is at a high level? I'm guessing you don't have a requirement to mix incompatible schemas on the same topic?

Reply:
In a high level we plan to send a json/avro(not yet decided) msg to a service using Kafka, that service handles that message and writes another message using kafka, which another service reads, and so on in a potential pipeline of services.

We will not mix different schema data types in the same topic.
All of the schema operations (create/delete/retrieve) will probably happen using different methods (due to permissions and other stuff), so I think we will only use the JsonSchema/AvroSchema classes and the encode/decode method to stay compatible with msg schema used by this/java pkg (first magic bit, then compression bit, than schema...)

Regarding the naming API, since you get the schema in the method I guess you can raise an error in case the strategy requires a name, or add an "avro_naming_strategy" and "json_naming_strategy" arguments to the serializer

Reply:
I suppose that some users will add an extra field to their json schemas that indicates the name of the schema, and they'd want the deserializer to use that. That means the user needs a way to customize how `JsonSchema`s are constructed, probably by telling the serializer how to pick a FQN for a new schema. The deserializer should also set the FQN appropriately on downloaded schemas, which is more tricky because it has to figure out which strategy was used to name the schema and whether it is able to mine the FQN out of that (or not).

I think this PR is fine as is. I don't have a requirement for naming strategies with json schema and it's going to require a bit more thought. In the short term the naming strategies ought to throw an error or at least log a warning if they are given a schema that doesn't have a FQN.